### PR TITLE
Added support for Ocean CD Create, update, patch, get, get all, Delet…

### PR DESCRIPTION
**Pull request by janetlinkiruba**
…e Strategy routes

Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiStrategy.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStrategy implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String>                               isSet;
+    private String                                    name;
+    private ApiStrategyCanary canary;
+    private Date                                      createdAt;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiStrategy.java:21]: Do we need these date fields?

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/Strategy.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class Strategy {
+    @JsonIgnore
+    private Set<String>                               isSet;
+    private String                                    name;
+    private StrategyCanary                            canary;
+    private Date                                      createdAt;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/Strategy.java:18]: Date fields needed?

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+            if (src.isPauseSet()) {
+                apiCanarySteps.setPause(toDal(src.getPause()));
+            }
+            if (src.isSetCanaryScaleSet()) {
+                apiCanarySteps.setSetWeight(src.getSetWeight());
+            }
+            if (src.isVerificationSet()) {
+                apiCanarySteps.setVerification(toDal(src.getVerification()));
+            }
+            if (src.isVerificationSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:157]: Typo.

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+        if (src != null) {
+            apiCanarySteps = new ApiCanarySteps();
+
+            if (src.isNameSet()) {
+                apiCanarySteps.setName(src.getName());
+            }
+            if (src.isPauseSet()) {
+                apiCanarySteps.setPause(toDal(src.getPause()));
+            }
+            if (src.isSetCanaryScaleSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:151]: Typo.

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+        ApiStepsSetCanaryScale apiStepsSetCanaryScale = null;
+
+        if (src != null) {
+            apiStepsSetCanaryScale = new ApiStepsSetCanaryScale();
+
+            if (src.isMatchTrafficWeightSet()) {
+                apiStepsSetCanaryScale.setMatchTrafficWeight(src.getMatchTrafficWeight());
+            }
+
+            if (src.isMatchTrafficWeightSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:188]: Typo

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+        if (src != null) {
+            CanarySteps.Builder canaryStepsBuilder = CanarySteps.Builder.get();
+
+            if (src.isNameSet()) {
+                canaryStepsBuilder.setName(src.getName());
+            }
+            if (src.isPauseSet()) {
+                canaryStepsBuilder.setPause(toBl(src.getPause()));
+            }
+            if (src.isSetCanaryScaleSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:348]: Typo

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+                canaryStepsBuilder.setSetWeight(src.getSetWeight());
+            }
+            if (src.isVerificationSet()) {
+                canaryStepsBuilder.setVerification(toBl(src.getVerification()));
+            }
+            if (src.isVerificationSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:354]: Typo

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Please review the comments and make appropriate fixes.

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiStrategy.java (hunk 1 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiStrategy.java:21]: yes these are part of response

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/Strategy.java (hunk 2 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/Strategy.java:18]: Yes these are part of response

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 3 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:157]: fixed and retested with example

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 4 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:151]: fixed and retested with example

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 5 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:188]: fixed and retested with example

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 6 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:348]: fixed and retested with example

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 7 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:354]: fixed and retested with example

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**